service/src/main/java/org/cbioportal/service/impl/GenesetCorrelationServiceImpl.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (c) 2016 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.math.NumberUtils;
+import org.apache.commons.math3.stat.correlation.SpearmansCorrelation;
+import org.cbioportal.model.CancerStudy;
+import org.cbioportal.model.Gene;
+import org.cbioportal.model.GenesetCorrelation;
+import org.cbioportal.model.GenesetGeneticData;
+import org.cbioportal.model.GeneGeneticData;
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.model.Sample;
+import org.cbioportal.model.GeneticProfile.DataType;
+import org.cbioportal.persistence.SampleListRepository;
+import org.cbioportal.service.GenesetService;
+import org.cbioportal.service.GenesetCorrelationService;
+import org.cbioportal.service.GenesetDataService;
+import org.cbioportal.service.GeneticDataService;
+import org.cbioportal.service.GeneticProfileService;
+import org.cbioportal.service.SampleService;
+import org.cbioportal.service.exception.GeneticProfileNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GenesetCorrelationServiceImpl implements GenesetCorrelationService {
+
+    @Autowired
+    private GeneticDataService geneticDataService;
+    @Autowired
+    private GenesetDataService genesetDataService;
+    @Autowired
+    private GeneticProfileService geneticProfileService;
+    @Autowired
+    private GenesetService genesetService;
+    @Autowired
+    private SampleService sampleService;
+    @Autowired
+    private SampleListRepository sampleListRepository;
```

Comment:
Done. Not sure why I used SampleListRepository...maybe SampleListService was not there yet....anyway nice catch.